### PR TITLE
Rust testing

### PR DESCRIPTION
**Pull request by droogmic**
Add more facilities for testing in rust.

### CODE REVIEW

File: rust/osm2lanes/src/lib.rs [hunk 1]
```diff
@@ -167,20 +167,27 @@ mod tests {
     use std::fs::File;
     use std::io::BufReader;
 
+    #[derive(Deserialize)]
+    #[serde(untagged)]
+    enum Rust {
```
Comment (dabreegster) [inline rust/osm2lanes/src/lib.rs:172]: Maybe `IncludeRustTest` or something would be easier to understand?

File: rust/osm2lanes/src/lib.rs [hunk 2]
```diff
@@ -167,20 +167,27 @@ mod tests {
...
+        Bool(bool),
+        Struct { separator: Option<bool> },
```
Comment (dabreegster) [inline rust/osm2lanes/src/lib.rs:174]: How about `IncludeWithOptions`?

File: rust/osm2lanes/src/lib.rs [hunk 3]
```diff
@@ -204,6 +211,32 @@ mod tests {
...
+        }
+        /// Is test enabled for separators, true by default
+        fn is_separators(&self) -> bool {
+            match self.rust {
+                None => true,
+                Some(Rust::Bool(_)) => unreachable!(),
+                Some(Rust::Struct { separator }) => separator.unwrap_or(true),
+            }
+        }
+        fn road(&self) -> Road {
```
Comment (dabreegster) [inline rust/osm2lanes/src/lib.rs:230]: Maybe `expected_output()`?

File: rust/osm2lanes/src/lib.rs [hunk 4]
```diff
@@ -204,6 +211,32 @@ mod tests {
                 println!("    {} = {}", k, v);
             }
         }
+        /// Is test enabled
+        fn is(&self) -> bool {
```
Comment (dabreegster) [inline rust/osm2lanes/src/lib.rs:215]: `is_enabled` and then nuke the docstring? :P

**Review by dabreegster — APPROVED** [review]

**Comment by droogmic** [comment]
As you noticed, I had a real problem with naming things when I was working on this.
agreed with all, will rework.

**Comment by dabreegster** [comment]
https://martinfowler.com/bliki/TwoHardThings.html :(

**Comment by droogmic** [comment]
I gave some alternative naming suggestions :P

File: rust/osm2lanes/src/lib.rs [hunk 5]
```diff
@@ -167,31 +167,48 @@ mod tests {
...
         description: Option<String>,
         // Data
         tags: Tags,
         output: Vec<Lane>,
+        // Skipping
+        rust: Option<RustTesting>,
+    }
+
+    impl DrivingSide {
+        /// Three-letter abbreviation
```
Comment (dabreegster) [inline rust/osm2lanes/src/lib.rs:194]: Gee, who wouldn't realize that? >_<

**Review by dabreegster — APPROVED** [review]
New names sound good!

**Pull request merged**